homeassistant/components/whirlpool/climate.py

```diff
@@ -110,8 +110,20 @@ def __init__(self, hass, said, name, backend_selector: BackendSelector, auth: Au
         self._aircon = Aircon(backend_selector, auth, said, self.async_write_ha_state)
 
         self.entity_id = generate_entity_id(ENTITY_ID_FORMAT, said, hass=hass)
-        self._attr_name = name if name is not None else said
```

Comment:
This will change the default visible name; is that what you want?

Reply:
Sorry about not understanding totally: Are you saying it will change it after a reboot? (not maintain the user modified name?)  I think it's ok for new users to have a slightly different name.... In the end they can change it to whatever they want.

Reply:
Here is how an entity's `name` parameter gets determined: https://github.com/home-assistant/core/blob/c0d5ceb18c9eb817125de0a88f202d990f7bbf79/homeassistant/helpers/entity.py#L320-L327

`_name` isn't a part of that. Am I missing where you set the entity name elsewhere?

Reply:
Looks like I messed up the has_entity_name - the climate sensor isn't setup that way :(
I've moved back to ._attr_name.  I think that is what we want.

Reply:
Since we set the device name = name AND has_entity_name = true, the generated name is the device name with no entity name -> same name as before adding DeviceInfo.